clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10736,8 +10737,16 @@ void SYCLPostLink::ConstructJob(Compilation &C, const JobAction &JA,
   ArgStringList CmdArgs;
 
   llvm::Triple T = getToolChain().getTriple();
+  const toolchains::SYCLToolChain &TC =
+      static_cast<const toolchains::SYCLToolChain &>(getToolChain());
+
+  // Handle -Xdevice-post-link
+  TC.TranslateTargetOpt(T, TCArgs, CmdArgs, options::OPT_Xdevice_post_link,
+                        options::OPT_Xdevice_post_link_EQ,
+                        JA.getOffloadingArch());
```

Comment:
The options passed by the user should override any options we set in the driver, so this should be moved to after `getTripleBasedSYCLPostLinkOpts`

Reply:
Ah....Ok....I just moved it to match the order in new-offloading model. I will move this down in both places. Thanks. Good catch.